src/PulseAudioManager.vala

```diff
@@ -42,50 +42,169 @@ public class Sound.PulseAudioManager : GLib.Object {
     private PulseAudio.GLibMainLoop loop;
     private bool is_ready = false;
     private uint reconnect_timer_id = 0U;
-    private Gee.HashMap<uint32, Device> input_devices;
-    private Gee.HashMap<uint32, Device> output_devices;
+    private Gee.HashMap<string, Device> input_devices;
+    private Gee.HashMap<string, Device> output_devices;
     public Device default_output { get; private set; }
     public Device default_input { get; private set; }
     private string default_source_name;
     private string default_sink_name;
     private Gee.HashMap<uint32, PulseAudio.Operation> volume_operations;
 
     private PulseAudioManager () {
-        
+
     }
 
     construct {
         loop = new PulseAudio.GLibMainLoop ();
-        input_devices = new Gee.HashMap<uint32, Device> ();
-        output_devices = new Gee.HashMap<uint32, Device> ();
+        input_devices = new Gee.HashMap<string, Device> ();
+        output_devices = new Gee.HashMap<string, Device> ();
         volume_operations = new Gee.HashMap<uint32, PulseAudio.Operation> ();
     }
 
     public void start () {
         reconnect_to_pulse.begin ();
     }
 
-    public void set_default_device (Device device) {
-        if (device.input) {
-            default_source_name = device.name;
-            var ope = context.set_default_source (device.name, null);
-            if (ope != null) {
-                PulseAudio.ext_stream_restore_read (context, ext_stream_restore_read_source_callback);
-            }
-        } else {
-            default_sink_name = device.name;
-            var ope = context.set_default_sink (device.name, null);
-            if (ope != null) {
-                PulseAudio.ext_stream_restore_read (context, ext_stream_restore_read_sink_callback);
+    public async void set_default_device (Device device) {
+        debug ("\n");
+        debug ("set_default_device: %s", device.id);
+        debug ("  %s", device.input? "input" : "output");
+        debug ("#1");
```

Comment:
I would remove all the debugs with "#", I don't think this adds anything worthy to the debug output.

Reply:
For me it made it easier to debug the steps, to see which steps where skipped for example, but I'll remove them.